src/material/paginator/paginator.ts

```diff
@@ -62,13 +62,18 @@ export class PageEvent {
   length: number;
 }
 
+/** The set of provided page size options to display to the user. */
+export type PageSizeOptions =
+  | readonly (number | {value: number; label: string})[]
+  | (number | {value: number; label: string})[];
+
 /** Object that can be used to configure the default options for the paginator module. */
 export interface MatPaginatorDefaultOptions {
   /** Number of items to display on a page. By default set to 50. */
   pageSize?: number;
 
   /** The set of provided page size options to display to the user. */
-  pageSizeOptions?: number[];
+  pageSizeOptions?: (number | {value: number; label: string})[];
```

Comment:
Shouldn't this and the one on `_MatPaginatorBase` use `PageSizeOptions`?

Reply:
Originally, the `MatPaginatorDefaultOptions` didn't support the readonly type, so I kept it as not supporting the readonly types.